fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/internal/DataLoader.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.providers.internal;
+
+import static com.netflix.spinnaker.security.AuthenticatedRequest.allowAnonymous;
+
+import com.netflix.spinnaker.fiat.providers.HealthTrackable;
+import com.netflix.spinnaker.fiat.providers.ProviderHealthTracker;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataLoader<T> implements HealthTrackable {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final Supplier<List<T>> loadingFunction;
+  private final ProviderHealthTracker healthTracker;
+
+  private AtomicReference<List<T>> cache = new AtomicReference<>();
+
+  public DataLoader(ProviderHealthTracker healthTracker, Supplier<List<T>> loadingFunction) {
+    this.healthTracker = healthTracker;
+    this.loadingFunction = loadingFunction;
+    getData();
```

Comment:
I wonder if there is a way to call this after Spring has instantiated beans, like on a Spring event?

Reply:
I changed it to fire on ContextRefreshedEvent